middleware/validateProductData.js

```diff
@@ -0,0 +1,47 @@
+const Validator = require('validator')
+const isEmpty = require('is-empty')
+
+function validateProductInput (data) {
+  const errors = {}
+  // Convert empty fields to an empty string so we can use validator functions
+  let { name, description, price, stock } = data
+  name = name || ''
+  description = description || ''
+  price = price || ''
+  stock = stock || ''
+
+  // Name checks
+  if (Validator.isEmpty(name)) {
+    errors.name = 'Name field is required'
+  }
+
+  // description checks
+  if (Validator.isEmpty(description)) {
+    errors.description = 'Description field is required'
+  }
+
```

Comment:
using the if else clause only returns a single validation error. to help with the front end i suggest we use multiple if statements rather than if else @DimejiAre